File: project/frontend/src/services/diagnosis.ts
import api from './api';

export interface DiagnosisLayerResult {
  score: number;
  issues: Array<{
    turn?: number;
    issue?: string;
    original?: string;
    suggested?: string;
    current_strategy?: string;
    suggested_strategy?: string;
  }>;
}

export interface DiagnosisResult {
  overall_score: number;
  psychology_layer: DiagnosisLayerResult;
  strategy_layer: DiagnosisLayerResult;
  script_layer: DiagnosisLayerResult;
  improvement_plan: string[];
}

export interface DiagnosisAnalyzeResponse {
  report_id: string;
  result: DiagnosisResult;
}

export interface DiagnosisReportItem {
  id: string;
  conversation_text: string;
  overall_score: number;
  result: DiagnosisResult;
  created_at: string;
}

export interface DiagnosisReportListResponse {
  items: DiagnosisReportItem[];
  total: number;
}

export const diagnosisService = {
  async analyze(conversationText: string): Promise<DiagnosisAnalyzeResponse> {
    const res = await api.post<DiagnosisAnalyzeResponse>('/diagnosis/analyze', {
      conversation_text: conversationText,
    });
    return res.data;
  },

  async getReports(page = 1, pageSize = 20): Promise<DiagnosisReportListResponse> {
    const res = await api.get<DiagnosisReportListResponse>('/diagnosis/reports', {
      params: { page, page_size: pageSize },
    });
    return res.data;
  },

  async getReport(reportId: string): Promise<DiagnosisReportItem> {
    const res = await api.get<DiagnosisReportItem>(`/diagnosis/reports/${reportId}`);
    return res.data;
  },
};

File: project/backend/app/schemas/diagnosis.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class DiagnosisAnalyzeRequest(BaseModel):
    conversation_text: str = Field(min_length=10)


class DiagnosisIssue(BaseModel):
    turn: int
    issue: str
    original: str | None = None
    suggested: str | None = None
    current_strategy: str | None = None
    suggested_strategy: str | None = None


class DiagnosisLayerResult(BaseModel):
    score: int
    issues: list[DiagnosisIssue]


class DiagnosisResult(BaseModel):
    overall_score: int
    psychology_layer: DiagnosisLayerResult
    strategy_layer: DiagnosisLayerResult
    script_layer: DiagnosisLayerResult
    improvement_plan: list[str]


class DiagnosisAnalyzeResponse(BaseModel):
    report_id: uuid.UUID
    result: DiagnosisResult


class DiagnosisReportResponse(BaseModel):
    id: uuid.UUID
    conversation_text: str
    overall_score: int
    result: dict
    created_at: datetime

    model_config = {"from_attributes": True}


class DiagnosisReportListResponse(BaseModel):
    items: list[DiagnosisReportResponse]
    total: int
